packages/prospectapi-common/src/lib.ts

```diff
@@ -221,3 +230,73 @@ export const deriveUrlMetadata = async (
     };
   }
 };
+
+/**
+ * use node fetch to hit the external metadata parser.
+ *
+ * broken out into a standalone function for easier mocking in tests.
+ *
+ * @param url
+ * @param metadataParserEnpointUrl
+ * @param metadataParserApiKey
+ * @returns JSON
+ */
+export const fetchUrlMetadata = async (
+  url: string,
+  metadataParserEnpointUrl: string,
+  metadataParserApiKey: string,
+): Promise<{ [key: string]: any } | null> => {
+  // Zyte-specific implementation
+  try {
+    const response = await fetch(metadataParserEnpointUrl, {
+      method: 'POST',
+      headers: {
+        'Content-Type': 'application/json',
+        Authorization: `Basic ${Buffer.from(
+          metadataParserApiKey + ':',
+        ).toString('base64')}`,
+      },
+      body: JSON.stringify({
+        url: url,
+        article: true,
+        articleOptions: {
+          extractFrom: 'httpResponseBody',
+        },
+        followRedirect: true,
+      }),
```

Comment:
One nice-to-have before we bring this to production would be to assign tags in the request so we can attribute cost and request volume in the Zyte console:

> tags
> object or null
> Assign arbitrary key-value pairs to the request that you can use for filtering in the Stats API.
> Keys must be strings. Values must be strings or null.
> For example: {"tags": {"foo": "bar", "baz": null}}.


If these are easily available, perhaps we can start with the environment and service name?